src/views/Fetchurl.vue

```diff
@@ -1,76 +1,189 @@
+              <v-btn
+                    id="copyText"
+                    class="text-no-wrap my-2"
+                    color="primary"
+                    type="submit"
+              >
+                  <span>Copy this URL to your clipboard</span>
+              </v-btn>
+            </form>
+            </div>
+          
+
+
+
+        </div>
+
+        <div v-if="failedCreate" class="d-flex">
+          <v-icon class="icon-green mr-2" :icon="mdiCheckCircle" />
+          <div>
+            <strong>
+              {{ responseFromGcp }}
+            </strong>
+          </div>
         </div>
-      </form>
-      
 
-      <div><span v-if="wasCreated">7-day download URL to deliver to user:</span></div>
-      <div><span v-if="wasCreated"> {{ download_url }} </span></div>
 
 
-      
+    </div>
+
   </div>
-  
+
+
+
 </template>
 
 <script>
 import Context from '@/mixins/Context'
-import {fetchUrl} from '@/utils'
+import {getCasLoginURL, getCasLogoutURL} from '@/api/user'
+import {fetchUrl} from '@/api/fetch-url'
+import Header1 from '@/components/utils/Header1.vue'
 
 export default {
-  name: 'fetchUrl',
+  name: 'Login',
   mixins: [Context],
   data: () => ({
     gs_source_url: undefined,
-    download_url: undefined,
-    isCreating: undefined,
-    wasCreated: undefined
+    responseFromGcp: undefined,
+    isRequesting: undefined,
+    wasCreated: undefined,
+    failedCreate: undefined
   }),
+  components: {Header1},
   methods: {
+    toFetchUrl() {
+      window.location.href = `http://localhost:8080/fetchurl`
+    },
+    casLogin() {
+      getCasLoginURL().then(data => window.location = data.casLoginURL)
+    },
+    logOut() {
+      getCasLogoutURL().then(data => window.location.href = data.casLogoutURL)
+    },
     fetchUrl() {
-      this.isCreating = true
-//      this.$announcer.polite('Fetching download url...')
-      this.download_url = fetchUrl(this.gs_source_url)
-      this.isCreating = false
-      this.wasCreated = true
+      this.isRequesting = true
+      fetchUrl(this.gs_source_url).then(data => {
+        this.isRequesting = false
+        if (data.status == "success") {
+          this.wasCreated = true
+          this.failedCreate = false
+        } else {
+          this.wasCreated = false
+          this.failedCreate = true
+        }
+        this.responseFromGcp = data.response
+      })
+      isRequesting = false
+    },
+    copyText() {
+      navigator.clipboard.writeText(this.responseFromGcp);
+    },
+      clearContents() {
+      this.gs_source_url = '';
+      this.responseFromGcp = '';
     }
   }
 }
 </script>
+
+
+<style scoped lang="scss">
+.page-fetch-url {
+  color: $color-off-black;
+  font-family: $body-font-family;
+  font-size: 14px;
+  font-weight: 300;
+  padding: 10px 20px;
+  .page-fetch-url-heading {
+    font-family: $body-font-family;
+    font-size: 23px;
+    font-weight: normal;
+    margin: 10px 0;
+  }
+  .fetch-url-label {
+    font-weight: 400;
+  }
+}
+</style>
```

Comment:
Since you're building this Vue 3 app from scratch, you can use Vue's [Composition API](https://vuejs.org/api/composition-api-setup).  DevOps is still behind the curve in this respect because we're most familiar with the old "Options API" syntax. Our two Vue3 apps (BOA and Ripley) were migrated from Vue 2 and inherited a bunch of the old syntax.  For the record, I do find the Composition API cleaner.